policy-controller/k8s/index/src/index.rs

```diff
@@ -201,15 +216,192 @@ impl Index {
 
     fn apply_route<R>(&mut self, route: R)
     where
-        R: ResourceExt,
+        R: ResourceExt + GetParentRefs + Clone + IntoRouteBinding,
         InboundRouteBinding: TryFrom<R>,
         <InboundRouteBinding as TryFrom<R>>::Error: std::fmt::Display,
     {
+        fn make_patch(name: &str, status: gateway::HttpRouteStatus) -> serde_json::Value {
+            serde_json::json!({
+                "apiVersion": "policy.linkerd.io/v1alpha1",
+                "kind": "HTTPRoute",
+                "name": name,
+                "status": status,
+            })
+        }
```

Comment:
hmm, it feels a bit weird to me that the indexer is what's responsible for actually constructing the JSON patches...my preference would be to send a more structured internal representation to the task that's actually responsible for  applying patches, and construct the JSON there, right before we send it over the wire.

if that complicates the implementation a lot, maybe it's not worth it, but it feels like a clearer separation of concerns to me...